sdks/uniswapx-sdk/integration/test/PriorityOrder.spec.ts

```diff
@@ -0,0 +1,723 @@
+      //const userSignature = await swapper._signTypedData(domain, types, values);
+      const cosignature = await cosigner.signMessage(
+        partialOrder.cosignatureHash(cosignerData)
+      );
+
+      const order = preBuildOrder
+        .cosignature(cosignature)
+        .cosignerData(cosignerData)
+        .build();
+
+      expect(order.info.deadline).to.eq(deadline);
+      expect(order.info.swapper).to.eq(swapperAddress);
+      expect(order.info.cosigner).to.eq(cosignerAddress);
+      expect(order.info.cosignature).to.eq(cosignature);
+      expect(order.info.nonce.toNumber()).to.eq(NONCE);
+
+      expect(order.info.input.token).to.eq(tokenIn.address);
+      expect(order.info.input.amount).to.eq(AMOUNT);
+
+      const builtOutput = order.info.outputs[0];
+
+      expect(builtOutput.token).to.eq(tokenOut.address);
+      expect(builtOutput.amount.eq(BigNumber.from(10).pow(17).mul(9))).to.be
+        .true;
+      expect(builtOutput.recipient).to.eq(swapperAddress);
+    });
+
+    // cosignature is only checked if current block < auctionStartBlock, so we set it in the future
+    it("reverts if cosignature is invalid", async () => {
+      const auctionStartBlock = BigNumber.from(block).add(3);
+      const deadline = await new BlockchainTime().secondsFromNow(1000);
+      const order = new PriorityOrderBuilder(
+        chainId,
+        reactor.address,
+        permit2.address
+      )
+        .cosigner(cosigner.address)
+        .auctionStartBlock(auctionStartBlock)
+        .baselinePriorityFeeWei(BigNumber.from(1))
+        .deadline(deadline)
+        .swapper(swapper.address)
+        .nonce(NONCE)
+        .input({
+          token: tokenIn.address,
+          amount: AMOUNT,
+          mpsPerPriorityFeeWei: BigNumber.from(0),
+        })
+        .output({
+          token: tokenOut.address,
+          amount: AMOUNT,
+          mpsPerPriorityFeeWei: BigNumber.from(1),
+          recipient: swapper.address,
+        })
+        .buildPartial();
+
+      const { domain, types, values } = order.permitData();
+      const signature = await swapper._signTypedData(domain, types, values);
+
+      const cosignerData = getCosignerData({
+        auctionTargetBlock: auctionStartBlock.sub(1),
+      });
+      const cosignerHash = order.cosignatureHash(cosignerData);
+      let cosignature = ethers.utils.joinSignature(
+        cosigner._signingKey().signDigest(cosignerHash)
+      );
+      const fullOrder = PriorityOrderBuilder.fromOrder(order)
+        .cosignerData({ auctionTargetBlock: auctionStartBlock.sub(2) })
+        .cosignature(cosignature)
+        .build();
+
+      await expect(
+        reactor
+          .connect(filler)
+          .execute(
+            { order: fullOrder.serialize(), sig: signature },
+            { maxPriorityFeePerGas: 3 }
+          )
+      ).to.be.revertedWithCustomError(reactor, "InvalidCosignature");
+    });
+
+    it("reverts if block is before auctionTargetBlock", async () => {
```

Comment:
What about a test for checking revert after expiry?

Reply:
chatted offline, will add in a followup pr later